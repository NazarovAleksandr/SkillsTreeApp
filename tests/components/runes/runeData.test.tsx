import * as React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import RuneData from '../../../src/components/runes/runeData';
import {
    IRune, Rarity, RuneTypes, RunePropertyTypes,
} from '../../../src/components/runes/models';

describe('RuneData', () => {
    test('Renders positive props', () => {
        const rune: IRune = {
            id: '1',
            image: 'url',
            rarity: Rarity.common,
            type: RuneTypes.attack,
            properties: [
                { name: RunePropertyTypes.Evade, value: 10 },
            ],
        };

        const component = shallow(<RuneData rune={rune} />);

        expect(component).toMatchSnapshot('positiveProps');
    });
    test('Renders negative props', () => {
        const rune: IRune = {
            id: '1',
            image: 'url',
            rarity: Rarity.common,
            type: RuneTypes.attack,
            properties: [
                { name: RunePropertyTypes.Evade, value: -10 },
            ],
        };

        const component = shallow(<RuneData rune={rune} />);

        expect(component).toMatchSnapshot('negativeProps');
    });
    test('Renders rare rarity', () => {
        const rune: IRune = {
            id: '1',
            image: 'url',
            rarity: Rarity.rare,
            type: RuneTypes.attack,
            properties: [
                { name: RunePropertyTypes.Evade, value: -10 },
            ],
        };

        const component = shallow(<RuneData rune={rune} />);

        expect(component).toMatchSnapshot('rareRarity');
    });
    test('Renders unique rarity', () => {
        const rune: IRune = {
            id: '1',
            image: 'url',
            rarity: Rarity.unique,
            type: RuneTypes.attack,
            properties: [
                { name: RunePropertyTypes.Evade, value: -10 },
            ],
        };

        const component = shallow(<RuneData rune={rune} />);

        expect(component).toMatchSnapshot('uniqueRarity');
    });
});
