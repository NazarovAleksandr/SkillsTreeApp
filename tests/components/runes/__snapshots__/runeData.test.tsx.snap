// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuneData Renders negative props: negativeProps 1`] = `
<div
  className="rune-data-container"
>
  <div
    className="main-data"
  >
    <div>
      <span
        className="title"
      >
        Type:
      </span>
       
      Attack
    </div>
    <div>
      <span
        className="title"
      >
        Rarity:
      </span>
       
      <span
        className="rune-rarity common"
      >
        Common
      </span>
    </div>
  </div>
  <div
    key="Evade-10"
  >
    <span
      className="title"
    >
      Evade
      :
    </span>
     
    <span
      className="property-value negative"
    >
      -10
    </span>
  </div>
</div>
`;

exports[`RuneData Renders positive props: positiveProps 1`] = `
<div
  className="rune-data-container"
>
  <div
    className="main-data"
  >
    <div>
      <span
        className="title"
      >
        Type:
      </span>
       
      Attack
    </div>
    <div>
      <span
        className="title"
      >
        Rarity:
      </span>
       
      <span
        className="rune-rarity common"
      >
        Common
      </span>
    </div>
  </div>
  <div
    key="Evade10"
  >
    <span
      className="title"
    >
      Evade
      :
    </span>
     
    <span
      className="property-value positive"
    >
      +10
    </span>
  </div>
</div>
`;

exports[`RuneData Renders rare rarity: rareRarity 1`] = `
<div
  className="rune-data-container"
>
  <div
    className="main-data"
  >
    <div>
      <span
        className="title"
      >
        Type:
      </span>
       
      Attack
    </div>
    <div>
      <span
        className="title"
      >
        Rarity:
      </span>
       
      <span
        className="rune-rarity rare"
      >
        Rare
      </span>
    </div>
  </div>
  <div
    key="Evade-10"
  >
    <span
      className="title"
    >
      Evade
      :
    </span>
     
    <span
      className="property-value negative"
    >
      -10
    </span>
  </div>
</div>
`;

exports[`RuneData Renders unique rarity: uniqueRarity 1`] = `
<div
  className="rune-data-container"
>
  <div
    className="main-data"
  >
    <div>
      <span
        className="title"
      >
        Type:
      </span>
       
      Attack
    </div>
    <div>
      <span
        className="title"
      >
        Rarity:
      </span>
       
      <span
        className="rune-rarity unique"
      >
        Unique
      </span>
    </div>
  </div>
  <div
    key="Evade-10"
  >
    <span
      className="title"
    >
      Evade
      :
    </span>
     
    <span
      className="property-value negative"
    >
      -10
    </span>
  </div>
</div>
`;
